src/transformers/modeling_attn_mask_utils.py

```diff
@@ -406,31 +406,18 @@ def _prepare_4d_attention_mask_for_sdpa(mask: torch.Tensor, dtype: torch.dtype,
         tgt_len (`int`):
             The target length or query length the created mask shall have.
     """
-    batch_size, key_value_length = mask.shape
+    _, key_value_length = mask.shape
```

Comment:
Can we change the input arg `mask: torch.Tensor` to `mask: Optional[torch.Tensor]` and `return None` immediately if `mask is None`? The docstring is not compliant with the actual input. (`mask (`torch.Tensor` or `None`):`)
Will it break the `is_tracing` check?

Reply:
@minostauros Yes indeed ideally we would want to do that. In practice, the calls to these functions in modeling files are always guarded by:
```python
if attention_mask is not None:
```

but we should IMO indeed accept `Optional[torch.Tensor]`. I'll leave that to an other PR.